include/Graph/Graph.hpp

```diff
@@ -0,0 +1,71 @@
+#pragma once
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+class Node {
+ public:
+  int abc;
+  std::vector<Node*> prevNodes;
+  std::vector<Node*> nextNodes;
```

Comment:
I think Node structure is too complex here, graph class are dealing with both prev and next Nodes using adjacency list